app/src/main/java/com/rud/movieapp/view/list/adapter/MovieListAdapter.kt

```diff
@@ -0,0 +1,74 @@
+package com.rud.movieapp.view.list.adapter
+
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.paging.PagedListAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.rud.domain.models.Movie
+import com.rud.movieapp.BR
+
+class MovieListAdapter(
+    private val retry: () -> Unit,
+    private val itemClickCallback: ItemClickCallback?
+) :
+    PagedListAdapter<Movie, RecyclerView.ViewHolder>(MovieDiffCallback) {
+
+    private val DATA_VIEW_TYPE = 1
+    private val FOOTER_VIEW_TYPE = 2
+
+    private var state = State.LOADING
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return if (viewType == DATA_VIEW_TYPE) MovieViewHolder.create(
+            parent
+        ) else LoadingViewHolder.create(retry, parent)
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        if (getItemViewType(position) == DATA_VIEW_TYPE) {
+            val binding = DataBindingUtil.bind<ViewDataBinding>(holder.itemView)
+            binding?.setVariable(BR.model, getItem(position))
```

Comment:
binding?.let {
                it.setVariable(BR.model, getItem(position))
                it.setVariable(BR.clickListener, itemClickCallback) 
            }

Reply:
Fixed.